operator/internal/controller/auto/workspace_controller.go

```diff
@@ -242,6 +242,20 @@ func (r *WorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	initializedV, ok := pod.Annotations[PodAnnotationInitialized]
 	initialized, _ := strconv.ParseBool(initializedV)
 	if !ok || !initialized {
+		l.Info("Running whoami to ensure authentication is setup correctly with the workspace pod")
+		_, err = wc.WhoAmI(ctx, &agentpb.WhoAmIRequest{})
+		if err != nil {
```

Comment:
Can we check the error code?  What if it a connection error vs auth error?

Consider calling `GetVersion` rather than `WhoAmI` if you want to isolate the GRPC aspects from the "state backend" aspect.

Reply:
The second iteration of this PR still calls `WhoAmI`. But now that we have structured errors returned, we can actually determine if it's a Pulumi access token error, or a k8s auth error.